great_expectations/experimental/datasources/pandas_filesystem_datasource.py

```diff
@@ -133,12 +78,24 @@ def add_csv_asset(
         asset = CSVAsset(
             name=name,
             regex=regex,
-            glob_directive=glob_directive,
             order_by=_batch_sorter_from_list(order_by or []),
             **kwargs,
         )
 
-        return self.add_asset(asset)
+        data_connector: DataConnector = FilesystemDataConnector(
+            datasource_name=self.name,
+            data_asset_name=name,
+            regex=regex,
+            base_directory=self.base_directory,
+            glob_directive=glob_directive,
+            data_context_root_directory=self.data_context_root_directory,
+        )
```

Comment:
Mostly a note for myself. This is where the `DataConnector` is being created.

Reply:
@Kilo59 Yes -- and I believe that it is in line with our idea that using this coding pattern we can reduce the combinatorial enumeration of leave `DataAsset` classes.  Thank you.